### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by OsvaldoPrajitno**
logout endpoint in the backend

### CODE REVIEW

File: backend/src/index.ts [hunk 1]
```diff
@@ -29,6 +36,29 @@ app.use(
   }),
 );
 
+app.post('/logout', async (req, res) => {
+  const sessionId = req.sessionID;
+  const querySnapshot = await getDocs(collection(db, 'sessions'))
```
Comment (bengodw) [inline backend/src/index.ts:35]: I don't think sessions need to be stored in the db but ig there isn't a problem with storing them here?

Comment (lachlanshoesmith) [reply backend/src/index.ts:35]: > I don't think sessions need to be stored in the db but ig there isn't a problem with storing them here?

sessions are generally stored in the db whereas jwts dont need to be, benefit of this is you can revoke a session at any time to invalidate it

File: backend/src/index.ts [hunk 2]
```diff
@@ -18,6 +18,13 @@ app.use(express.json());
 app.use(cors());
 app.use(
   session({
+    // TODO: uncomment the secure 
+    // secure: true,
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: i recommend we instead use `process.env.NODE_ENV` to determine whether we're in production/development and toggle this flag accordingly so we don't forget abt it

File: backend/src/index.ts [hunk 3]
```diff
@@ -18,6 +18,13 @@ app.use(express.json());
...
+    cookie: {
+      sameSite: "lax",
+      httpOnly: true,
+      maxAge: 604800000
+    },
     secret: process.env.SESSION_SECRET as string,
     saveUninitialized: false,
     /* I'm not entirely sure of the effects,
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: it's `true` by default, i think you can remove this line

**Review by lachlanshoesmith — CHANGES_REQUESTED** [review]

File: backend/src/index.ts (hunk 2 above)
Comment (OsvaldoPrajitno) [reply backend/src/index.ts:22]: would this be using another env variable like DEV_STATUS or is there already somethin we can use?

**Review by lachlanshoesmith — APPROVED** [review]

**Pull request merged**